### PR TITLE
[Sylius][ResourceBundle] Fix mapping with interface

### PR DESCRIPTION
**Pull request by gperdomor**
| Q               | A
| --------------- | ---
| Bug fix?        |yes
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | #4447
| License         | MIT
| Doc PR          | -

Remove <code>ResolveDoctrineTargetEntitiesPass</code> and handle interfaces in <code>RegisterResourcesPass</code>

### CODE REVIEW

**Comment by pjedrzejewski** [comment]
Looks good! :+1: Could you take a look at the failing build?

File: src/Sylius/Bundle/ResourceBundle/DependencyInjection/Compiler/RegisterResourcesPass.php [hunk 1]
```diff
@@ -33,8 +34,18 @@ public function process(ContainerBuilder $container)
         $resources = $container->getParameter('sylius.resources');
         $registry = $container->getDefinition('sylius.resource_registry');
 
+        $interfaces = [];
+
         foreach ($resources as $alias => $configuration) {
             $registry->addMethodCall('addFromAliasAndConfiguration', [$alias, $configuration]);
+
+            if (array_key_exists('interface', $configuration['classes'])) {
```
Comment (michalmarcinkowski) [inline src/Sylius/Bundle/ResourceBundle/DependencyInjection/Compiler/RegisterResourcesPass.php]: `if (isset($configuration['classes']['interface']))`

File: src/Sylius/Bundle/ResourceBundle/DependencyInjection/Compiler/RegisterResourcesPass.php [hunk 2]
```diff
@@ -33,8 +34,18 @@ public function process(ContainerBuilder $container)
...
+                $alias = explode('.', $alias);
```
Comment (michalmarcinkowski) [inline src/Sylius/Bundle/ResourceBundle/DependencyInjection/Compiler/RegisterResourcesPass.php]:
```
if (!isset($alias[0]) || !isset($alias[1])) {
    throw new ...
}
```

**Comment by gperdomor** [comment]
@pjedrzejewski honestly i don't know why fails. I don't know why <code>doctrine.orm.listeners.resolve_target_entity</code> is missing in the container

**Comment by pamil** [comment]
@gperdomor awesome job! :tada: I'm hacking on your code now and hopefully I'll make the tests pass :)

**Comment by pamil** [comment]
It looks like it works, I've sent a [pull request](https://github.com/gperdomor/Sylius/pull/2) to this PR's branch.

**Comment by gperdomor** [comment]
@pamil i merge some of your changes made in your PR, but not work :(

**Pull request closed without merging**